eng/pipelines/templates/jobs/build-cli.yml

```diff
@@ -89,14 +89,37 @@ jobs:
       - bash: dotnet nuget add source --name dotnet8 https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet8/nuget/v3/index.json
         displayName: Add internal dotnet nuget feed
 
-      - template: /eng/pipelines/templates/steps/azd-login.yml
-        parameters:
-          AzdDirectory: cli/azd
+      - task: AzureCLI@2
+        condition: and(succeeded(), ne(variables['Skip.LiveTest'], 'true'))
+        inputs:
+          azureSubscription: azd-service-connection
+          scriptType: pscore
+          scriptLocation: inlineScript
+          inlineScript: |
+            $subscriptionId = (az account show -o json | ConvertFrom-Json -AsHashtable)['id']
+
+            Write-Host "##vso[task.setvariable variable=AzureSubscriptionClientId;issecret=false]$($env:AZURESUBSCRIPTION_CLIENT_ID)"
+            Write-Host "##vso[task.setvariable variable=AzureSubscriptionTenantId;issecret=false]$($env:AZURESUBSCRIPTION_TENANT_ID)"
+            Write-Host "##vso[task.setvariable variable=AzureSubscriptionServiceConnectionId;issecret=false]$($env:AZURESUBSCRIPTION_SERVICE_CONNECTION_ID)"
+            Write-Host "##vso[task.setvariable variable=SubscriptionId;issecret=false]$($subscriptionId)"
+        displayName: Configure OIDC Authentication
```

Comment:
This should be in a template and put in https://github.com/Azure/azure-dev/blob/main/eng/pipelines/templates/steps/template-test-run-job.yml